parlai/core/agents.py

```diff
@@ -284,7 +284,10 @@ def load_agent_module(opt):
         return None
 
 def get_agent_module(dir_name):
-    if ':' in dir_name:
+    if '.' in dir_name:
+        module_name = dir_name.split(':')[0]
+        class_name = dir_name.split(':')[-1]
+    elif ':' in dir_name:
```

Comment:
what broke here?

is `len(dir_name.split(':')) > 1` when `'.' in dir_name`?

Reply:
I tried using an agent from parlai_external, which i had to specify as `-m "parlai_external.agents.mlb.mlb:MlbAgent"`

Reply:
shouldn't the module/class-name code below work for that? it looks like in both cases you should get `module_name = parlai_external.agents.mlb.mlb` and `class_name = MlbAgent`, right?

Reply:
You know, I could've sworn this wasn't working for me before, but it does in fact work without my addition. Will remove in the next commit